ui/txs/noves/useDescribeTxs.tsx

```diff
@@ -0,0 +1,89 @@
+import type { UseQueryResult } from '@tanstack/react-query';
+import _ from 'lodash';
+
+import type { NovesDescribeTxsResponse } from 'types/api/noves';
+import type { Transaction } from 'types/api/transaction';
+
+import type { ResourceError } from 'lib/api/resources';
+
+export interface DescribeTxs {
+  txHash: string;
+  type?: string;
+  description?: string;
+  isLoading: boolean;
+  enabled: boolean;
+}
+
+export interface TransactionWithTranslate extends Transaction {
+  translate?: DescribeTxs;
+}
+
+export default function useDescribeTxs(items: Array<Transaction> | undefined, translateEnabled: boolean) {
+  const txsHash = items?.map(i => i.hash);
+  const txsChunk = _.chunk(txsHash || [], 10);
+
+  const txsParams = Array(5).fill(undefined).map((_, i) => txsChunk[i]);
+  const txsQuerys = useFetchTxs(txsParams, translateEnabled);
+
```

Comment:
I think if the translation is not enabled, we should do an early return of unchanged data

Reply:
This issue has not been addressed yet. Even if the feature is disabled, some unnecessary computation will still occur and it will create 5 unnecessary hanging queries. 

To improve performance and memory consumption, I suggest rewriting this hook as follows:

```js
import useApiFetch from 'lib/api/useApiFetch';
import { useQuery } from '@tanstack/react-query';
import { getResourceKey } from 'lib/api/useApiQuery';

export default function useDescribeTxs(items: Array<Transaction> | undefined, viewAsAccountAddress: string | undefined) {

  const apiFetch = useApiFetch();

  const describeQuery = useQuery({
    // in queryKey we should probably also pass the page number so requests are properly cached
    // or use for example { firstHash, lastHash } props
    queryKey: [ 'noves_describe_txs', { viewAsAccountAddress } ],
    queryFn: async() => {
      // prepare data for Noves API call
      // ...
      const txChunks: Array<Array<string>> = [];

      const queries = txChunks.map((hashes) => {
        if (hashes.length === 0) {
          return Promise.resolve([]);
        }

        return apiFetch('noves_describe_txs', {
          queryParams: {
            viewAsAccountAddress,
            hashes,
          },
        });
      });

      return Promise.all(queries);
    },
    select: (data) => {
      // do all data manipulation here, like flattening and mapping
    },
    enabled: translateEnabled,
  });

  if (!translateEnabled) {
    return items;
  }

  return React.useMemo(() => {
    // modify items with translation data and isLoading flag
  }, []);

}

```